pkg/kv/kvserver/replica_rate_limit.go

```diff
@@ -32,9 +32,20 @@ func (r *Replica) maybeRateLimitBatch(ctx context.Context, ba *kvpb.BatchRequest
 		return nil
 	}
 
+	var info tenantcostmodel.BatchInfo
+	for i := range ba.Requests {
+		req := ba.Requests[i].GetInner()
+		if !kvpb.IsReadOnly(req) {
```

Comment:
The RU model only includes a subset of batch requests, whereas here, we're counting everything, which is desired outcome for eCPU models. What if the tenant is still using the RU model (but the limiter code here has already been updated for the eCPU model)?

Reply:
This is the KV tenant rate limiter, which is not related to the customer's model. It exists to prevent each KV node from being overloaded by a single tenant. There's a separate issue to convert this code to use the eCPU model, since it's more accurate. However, even before that happens, it's OK to loosen this code up a bit to include all requests, since none of this impacts customer costs.